package.json

```
@@ -1,23 +1,26 @@
 {
   "name": "path2d-polyfill",
-  "version": "1.2.3",
+  "version": "2.0.0-beta.0",
   "description": "Polyfills Path2D api for canvas rendering",
   "scripts": {
     "build": "rm -rf dist/* && rollup -c",
     "start": "rollup -c -w",
-    "lint": "eslint src test",
-    "test": "nyc --reporter=html --reporter=text mocha",
+    "lint": "eslint .",
+    "check-types": "tsc --noEmit",
+    "test": "ts-mocha test/*.spec.ts",
+    "test:watch": "ts-mocha --watch test/*.spec.ts",
+    "test:coverage": "nyc --reporter=html --reporter=text --reporter=text-summary yarn test",
     "format:check": "prettier --check \"./**\"",
     "format:write": "prettier --write \"./**\"",
     "prepare": "husky install",
     "release": "release-it"
   },
+  "browser": "dist/path2d-polyfill.min.js",
+  "module": "dist/path2d-polyfill.esm.js",
+  "main": "dist/path2d-polyfill.cjs.js",
```

It seems like `dist/path2d-polyfill.esm.js` is being built for `Node` (if I understand the Rollup config correctly), but as far as I know Node.js doesn't look at `module`, but WebPack does. So this will end up serving the Node.js build to browsers, and not to ESM-capable Node.js.

There are more information about packaging using ESM for Node.js here: https://nodejs.org/api/packages.html#dual-commonjses-module-packages

Personally, I would recommend to just ship CJS for Node.js (or potentially only ESM). That way you don't have the "dual package hazard" mentioned in the link above, and I also don't think that there is any benefit to shipping an esm file as well?

So in that case it could look something like:

```suggestion
  "exports": {
    ".": {
      "node": "dist/path2d-polyfill.cjs.js",
      "browser": "dist/path2d-polyfill.min.js"
    }
  }
```

The `browser` condition is supported by webpack, esinstall, and wmr. But could potentially be replaced with `default`, but I'm not sure if browser should be the default 🤔

Even more information is available here: https://webpack.js.org/guides/package-exports/

---

you're right that .esm builds with the `node` target, but that was unintentional. The intention was to build it for browsers so that webpack can pick it up ... and the idea about that is that whoever webpacks the code can get untranspiled and unminified code in order to bundle it with more control. I'll consider your suggestion.